compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/declaration/FirOverrideChecker.kt

```diff
@@ -130,6 +130,20 @@ object FirOverrideChecker : FirClassChecker() {
         }
     }
 
+    // See [DescriptorVisibilities.compare]
+    // If either one comes from Java, using the regular [Visibilities#compareTo], which uses enum ordinal, will return `null`, since there
+    // is no matched visibility. [JavaVisibilities] has Java-specific visibilities, along with their own [compareTo]. Hence, try comparison
```

Comment:
Which one among the tests you fixed has a Java-specific visibility?

Reply:
No, Java-specific ones were incorrectly reported as an error without this treatment.